src/main/java/org/mifos/identityaccountmapper/service/AccountLookupService.java

```diff
@@ -0,0 +1,47 @@
+package org.mifos.identityaccountmapper.service;
+
+import com.google.gson.Gson;
+import org.mifos.identityaccountmapper.data.BeneficiaryDTO;
+import org.mifos.identityaccountmapper.domain.IdentityDetails;
+import org.mifos.identityaccountmapper.domain.PaymentModalityDetails;
+import org.mifos.identityaccountmapper.repository.ErrorTrackingRepository;
+import org.mifos.identityaccountmapper.repository.MasterRepository;
+import org.mifos.identityaccountmapper.repository.PaymentModalityRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AccountLookupService {
+    private final MasterRepository masterRepository;
+    private final ErrorTrackingRepository errorTrackingRepository;
+    private final PaymentModalityRepository paymentModalityRepository;
+    private final SendCallbackService sendCallbackService;
+    private static final Logger logger = LoggerFactory.getLogger(AccountLookupService.class);
+
+    @Autowired
+    public AccountLookupService(MasterRepository masterRepository, ErrorTrackingRepository errorTrackingRepository,
+                                           PaymentModalityRepository paymentModalityRepository, SendCallbackService sendCallbackService){
+        this.masterRepository = masterRepository;
+        this.errorTrackingRepository = errorTrackingRepository;
+        this.paymentModalityRepository = paymentModalityRepository;
+        this.sendCallbackService = sendCallbackService;
+    }
+    @Async("asyncExecutor")
+    public void accountLookup(String callbackURL,String payeeIdentity){
+        try {
+            IdentityDetails identityDetails = masterRepository.findByPayeeIdentity(payeeIdentity);
+            PaymentModalityDetails paymentModalityDetails = paymentModalityRepository.findByMasterId(identityDetails.getMasterId());
+            sendAccountLookupCallback(new BeneficiaryDTO(identityDetails.getPayeeIdentity(),paymentModalityDetails.getModality(),paymentModalityDetails.getDestinationAccount(),paymentModalityDetails.getInstitutionCode()),callbackURL );
+        }catch (Exception e){
+            logger.error(e.getMessage());
+        }
+    }
+
+    private void sendAccountLookupCallback(BeneficiaryDTO accountDetails, String callbackURL){
+        Gson gson = new Gson();
```

Comment:
use Jackson instead of Gson. and create a single bean and auto wire wherever needed